products/healthcare/api.yaml

```diff
@@ -540,3 +540,67 @@ objects:
         'Creating a HL7v2 Store':
           'https://cloud.google.com/healthcare/docs/how-tos/hl7v2'
       api: 'https://cloud.google.com/healthcare/docs/reference/rest/v1/projects.locations.datasets.hl7V2Stores'
+  - !ruby/object:Api::Resource
+    name: 'ConsentStore'
+    kind: "healthcare#consentStore"
+    base_url: '{{dataset}}/consentStores?consentStoreId={{name}}'
+    self_link: '{{dataset}}/consentStores/{{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    min_version: beta
+    description: |
+      The Consent Management API is a tool for tracking user consents and the documentation associated with the consents. 
+    parameters:
+      - !ruby/object:Api::Type::ResourceRef
+        name: 'dataset'
+        description: |
+          Identifies the dataset addressed by this request. Must be in the format
+          'projects/{project}/locations/{location}/datasets/{dataset}'
+        required: true
+        input: true
+        resource: 'Dataset'
+        imports: 'selfLink'
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The resource name for the ConsentStore.
+
+          ** Changing this property may recreate the Consent store (removing all data) **
```

Comment:
This note is implied by the field being `input: true`, which marks the generated Terraform schema field as ForceNew which triggers recreate if changed